roles/backend/templates/backend-listener-deployment.yaml

```
@@ -24,6 +24,15 @@ spec:
         deployment: backend-listener
         threescale_component: backend
         threescale_component_element: listener
+{% if listener.marin3r.enabled == true %}
+        marin3r.3scale.net/status: "enabled"
+{% endif %}
```

You actually need to explicitly set the label to `disabled` when `listener.marin3r.enabled == false`, otherwise due to the patch behaviour the annotation won't be removed and marin3r will be still enabled. In fact, it happens the same with the annotations, but in that case is not a problem because the whole activation is controlled by this label.

```suggestion
{% if listener.marin3r.enabled == true %}
        marin3r.3scale.net/status: "enabled"
{% else %}
        marin3r.3scale.net/status: "disabled"
{% endif %}
```

---

so true, nice catch!

---

Fixed https://github.com/3scale/saas-operator/pull/19/commits/1719daf410e2d8b07a735412a3035716f65a6b36